src/idiom173ex.fz

```diff
@@ -0,0 +1,30 @@
+# Example that returns the formatted string instead of printing it directly
+
+format(x T : integer)  String =>
+
+    str:= x.as_string
+
+    len := str.byte_length
+
+    first :=
+        if x < T.zero && ((len%3) = 1) then 4   # avoid seperator right after minus
+        else if (len%3) > 0 then len%3 else 3   # determin size of first group
+    
+    groups(start i32, str String) String =>
+        if start > (str.byte_length - 3)
+        then ""
+        else "," + (str.substring start (start + 3)) + groups (start + 3) str
+
+    ((str.substring (0) (first)) + (groups first str))
```

Comment:
The code in `idiom173.fuzion` and `idiom173ex.fz` shoud essentially be the same, i.e., the first should be only the `format` function, while the second should call this with sample arguments and output the results.

Reply:
Would you recommend to leave out the version in `idiom173.fuzion` completely?
The reason I kept it, is because it does the task of the idiom and I thought it might be more efficient to print the substrings directly instead of building a new string recursively. Though I'm not sure now if it makes much difference.

The version in `idiom173ex.fz` is more useful in  general I think, because it returns the string instead of printing it.